packages/integration-tests/src/tests/api/hook/hook.data.hook.trigger.test.ts

```diff
@@ -0,0 +1,341 @@
+import {
+  RoleType,
+  hookEventGuard,
+  hookEvents,
+  jsonGuard,
+  managementApiHooksRegistration,
+  type Hook,
+  type Role,
+} from '@logto/schemas';
+import { z } from 'zod';
+
+import { authedAdminApi } from '#src/api/api.js';
+import { createResource } from '#src/api/resource.js';
+import { createScope } from '#src/api/scope.js';
+import {
+  OrganizationApiTest,
+  OrganizationRoleApiTest,
+  OrganizationScopeApiTest,
+} from '#src/helpers/organization.js';
+import { UserApiTest, generateNewUser } from '#src/helpers/user.js';
+import { generateName, waitFor } from '#src/utils.js';
+
+import WebhookMockServer from './WebhookMockServer.js';
+import {
+  organizationDataHookTestCases,
+  organizationRoleDataHookTestCases,
+  organizationScopeDataHookTestCases,
+  roleDataHookTestCases,
+  scopesDataHookTestCases,
+  userDataHookTestCases,
+} from './test-cases.js';
+
+const mockHookResponseGuard = z.object({
+  signature: z.string(),
+  payload: z.object({
+    event: hookEventGuard,
+    createdAt: z.string(),
+    hookId: z.string(),
+    body: jsonGuard.optional(),
+    method: z
+      .string()
+      .optional()
+      .transform((value) => value?.toUpperCase()),
+    _matchedRoute: z.string().optional(),
+  }),
+});
+
+type MockHookResponse = z.infer<typeof mockHookResponseGuard>;
+
+describe('Trigger DataHook through management API', () => {
```

Comment:
```suggestion
describe('trigger data hook through Management API', () => {
```

Reply:
actually no need for this layer of `describe`? you can write `beforeAll` and `afterAll` in the outmost scope.